Server/src/Server.Infrastructure/Persistence/QueryHandlers/GetBooksWithMostReservationsHandler.cs

```diff
@@ -26,14 +30,17 @@ public GetBooksWithMostReservationsHandler(BookBookDbContext dbContext)
             .OrderByDescending(g => g.ReservationCount)
             .Take(3)
             .ToListAsync(cancellationToken);
-
         var books = await _dbContext.Books
             .Where(x => booksWithMostReservations.Select(r => r.BookId).Contains(x.Id))
             .ToListAsync(cancellationToken);
 
         if (books.Count < 3)
         {
-            books = await _dbContext.Books.Take(3).ToListAsync(cancellationToken);
+            return await _dbContext.Books
+                .OrderBy(b => Guid.NewGuid())
+                .Take(3)
+                .ProjectTo<BookViewModel>(_mapper.ConfigurationProvider)
+                .ToListAsync(cancellationToken);
         }
 
```

Comment:
Tak tylko na szybko, może by tu dodać Cache? bo to też tak przy każdym otwarciu strony będzie leciało. Ale to nie mus, zależy jak tam ci się chce xd

Reply:
jest cache, w controllerze, bardziej pasuje do warstwy api